lib/firmware/ethereum.c

```diff
@@ -541,7 +583,11 @@ static void layoutEthereumFee(const uint8_t *value, uint32_t value_len,
  */
 
 static bool ethereum_signing_check(EthereumSignTx *msg) {
-  if (!msg->has_gas_price || !msg->has_gas_limit) {
+  if (!msg->has_gas_limit) {
+    return false;
+  }
+
+  if (!(msg->has_gas_price || msg->has_max_fee_per_gas)) {
```

Comment:
As I understand this logic, a tx has to have has_gas_price or has_max_fee_per_gas or both?

Reply:
That's right, tx has to have either a gas price set (non-EIP-1559) or max_fee_per_gas (EIP-1559.)